### DISCUSSION TITLE
Convert from Impulse response to S-parameter

### DISCUSSION THREAD

**Discussion by ZiadHatab** [scikit-rf/scikit-rf#1079] [Q&A] [answered]
Hi!

I don't use the time domain function very often. I calculated the impulse response with `rf.impulse_response()` and post-processed the response and now I want to convert back to the frequency domain. Is there a built-in function for this? 

Thanks!

**Comment by mhuser** [comment, marked as answer]
Hello @ZiadHatab

To my knowledge there is unfortunately no such built-in function.
In theory one can use FFT to turn back to frequency domain but it is not trivial because of padding, windowing, odd or even number of points, etc.

There are [time gating](https://scikit-rf.readthedocs.io/en/latest/examples/networktheory/Time%20Domain.html) function that does transformation to time-domain and back to frequency-domain, but they cannot take a raw time-domain input.

**Reply by ZiadHatab** [reply]
@mhuser 

Thanks for looking into this. I ended up just calculating the ifft myself and fft with Numpy. But you're right, I guess it's difficult to generalize a reverse operation to any processed signal.
